packages/nimbus/src/components/data-table/components/data-table.row.tsx

```diff
@@ -0,0 +1,289 @@
+import { forwardRef, useState } from "react";
+import { Row as RaRow, Cell as RaCell } from "react-aria-components";
```

Comment:
We should use the built-in collection optimizations from react-aria. [See example from the table content docs](https://react-spectrum.adobe.com/react-aria/Table.html#content)
```suggestion
import { Row as RaRow, Cell as RaCell, Collection as RaCollection } from "react-aria-components";
```